e2e/testcafe-devextreme/tests/editors/checkBox/common.ts

```diff
@@ -1,49 +1,79 @@
 import { createScreenshotsComparer } from 'devextreme-screenshot-comparer';
 import { Selector } from 'testcafe';
+import Guid from 'devextreme/core/guid';
 import url from '../../../helpers/getPageUrl';
 import { createWidget } from '../../../helpers/createWidget';
 import {
   appendElementTo,
-  insertStylesheetRulesToPage, setStyleAttribute,
+  insertStylesheetRulesToPage,
+  setClassAttribute,
+  setStyleAttribute,
 } from '../../../helpers/domUtils';
-import { testScreenshot } from '../../../helpers/themeUtils';
+import { getFullThemeName, testScreenshot } from '../../../helpers/themeUtils';
+
+const valueModes = [false, true, undefined];
 
 const CHECKBOX_CLASS = 'dx-checkbox';
+const READONLY_STATE_CLASS = 'dx-state-readonly';
+const DEFAULT_STATE_CLASS = '';
+const ACTIVE_STATE_CLASS = 'dx-state-active';
+const HOVER_STATE_CLASS = 'dx-state-hover';
+const FOCUSED_STATE_CLASS = 'dx-state-focused';
+const DISABLED_STATE_CLASS = 'dx-state-disabled';
 
-fixture.disablePageReloads`CheckBox`
+fixture.disablePageReloads`CheckBox render`
   .page(url(__dirname, '../../container.html'));
 
-[false, true].forEach((isColumnCountStyle) => {
-  test(`Render ${!isColumnCountStyle ? 'default' : 'with column-count style on container'}`, async (t) => {
```

Comment:
The initial test was written to fix an issue in a specific scenario - https://github.com/DevExpress/DevExtreme/pull/22051. After modifying the existing test, it is necessary to ensure that the new test continues to cover the scenario as well as the previous one did.

Reply:
You're right. I found also some issues with the screenshots too when I mixed them. I already left the existing test and created a new one.